app/logical/sources/alternates/furaffinity.rb

```diff
@@ -13,11 +13,36 @@ def domains
       end
 
       def parse
+        # Add gallery link, parsed from direct link
         if @url =~ IMAGE_TO_ARTIST
           @gallery_url = "https://www.furaffinity.net/user/#{$1}/"
-          @direct_url = @url
         end
       end
+
+      def original_url
+        # Handle old CDN or old broken CDN
+        if ["d.facdn.net", "d2.facdn.net"].include?(@parsed_url.host)
+          @parsed_url.host = "d.furaffinity.net"
+          @url = @parsed_url.to_s
```

Comment:
@url = @parsed_url.to_s seems to happen very often. It only needs to be done once, at the end of the method. @url = @parsed_url.to_s automatically returns the result of the asignment.

Even better would be if that happends somewhere else automatically, mabye in the base class or something. But theres no need to do that now if you don't feel like it.

Reply:
Hmm, yeah, it could be done in a wrapper method or something, but that does seem like it might be a tricky reworking of things? Possibly, baving a base method which returned `@parsed_url.to_s` instead of `@url` might prevent any other source handler doing string based url changes.
I can imagine how it could be done in different languages, but I'm not confident enough in Ruby to be sure the best way here. I suspect it would be more lines than just having it at the end of this method though, and potentially more confusion.

I'll just move it to the end in the handlers which do this often

Reply:
Yeah, it's fine like this